invenio_access/manage.py

```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015 CERN.
+#
+# Invenio is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+#
+# In applying this licence, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Perform access operations."""
+
+from __future__ import print_function
+
+from sys import stderr
+
+from invenio.ext.script import Manager
+
+from invenio.ext.sqlalchemy import db
+
+from invenio_accounts.models import User
+
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.orm.exc import NoResultFound
+
+from .models import AccROLE, UserAccROLE
+
+manager = Manager(usage=__doc__)
+
+__MSG_FAILED_TO_ADD_ROLE = 'failed to add a role:'
+__MSG_FAILED_TO_REMOVE_ROLE = 'failed to remove a role:'
+__MSG_FAILED_TO_GET_ROLES_FOR_USER = 'failed to get roles for user:'
+
+
+@manager.command
+@manager.option('--email', '-e', dest='email')
+@manager.option('--role_name', '-e', dest='role_name')
+def add(email, role_name):
+    """Add a role for a user."""
+    user = __get_user(email, __MSG_FAILED_TO_ADD_ROLE)
+    role = __get_role(role_name, __MSG_FAILED_TO_ADD_ROLE)
+
+    user_role = UserAccROLE()
+    user_role.id_user = user.id
+    user_role.id_accROLE = role.id
+
+    try:
+        db.session.begin(subtransactions=True)
+        db.session.add(user_role)
+        db.session.commit()
+        db.session.commit()
+    except IntegrityError:
+        db.session.rollback()
+        print('{0} has already been assigned {1} role'.format(email, role_name))
+        exit(0)
+    except Exception as e:
+        db.session.rollback()
+        print(__MSG_FAILED_TO_ADD_ROLE, e, file=stderr)
+        exit(1)
+
+    print('successfully added role {0} for user {1}'.format(
+        role.name, user.email))
+
+
+@manager.command
+@manager.option('--email', '-e', dest='email')
+@manager.option('--role_name', '-e', dest='role_name')
+def remove(email, role_name):
+    """Remove a role from a user."""
+    user = __get_user(email, __MSG_FAILED_TO_REMOVE_ROLE)
+    role = __get_role(role_name, __MSG_FAILED_TO_REMOVE_ROLE)
+
+    try:
+        user_role = UserAccROLE.get(user.id, role.id)
+        db.session.begin(subtransactions=True)
```

Comment:
you can use `with db.session.begin_nested():` (http://docs.sqlalchemy.org/en/rel_1_0/orm/session_transaction.html#using-savepoint)